solr/core/src/test/org/apache/solr/metrics/SolrMetricManagerTest.java

```diff
@@ -390,6 +394,32 @@ public void testObservableDoubleUpDownCounter() {
     assertEquals(-10.1, actual.getDataPoints().getFirst().getValue(), 1e-6);
   }
 
+  @Test
+  public void testMetricExporter() throws Exception {
+    LongCounter counter =
+        metricManager.longCounter(METER_PROVIDER_NAME, "my_counter", "desc", null);
+    counter.add(5);
+    counter.add(3);
+    InMemoryMetricExporter exporter = (InMemoryMetricExporter) metricManager.getMetricExporter();
+
+    // Wait 3 seconds for the exporter to push metrics
+    Thread.sleep(3000);
```

Comment:
It's better to use RetryUtil or similar technique instead of sleep.  Sleep is tests for large codebases like Solr is rather evil.